### PR TITLE
Don't drain etcd nodes

### PR DESCRIPTION
**Pull request by jakefhyde**
Draining etcd nodes causes issues in RKE2, where some pods are unhealthy. This PR never drains etcd nodes, which is the current approach in RKE1.
Related Issue: #35274

### CODE REVIEW

**Review by jiaqiluo — DISMISSED** [review]
LGTM

File: pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go [hunk 1]
```diff
@@ -303,6 +303,10 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 		if !machinePool.DrainBeforeDelete {
 			machineSpecAnnotations[capi.ExcludeNodeDrainingAnnotation] = "true"
 		}
+		if machinePool.EtcdRole {
+			logrus.Warnf("Ignoring drain before delete for pool [%s] with etcd role", machinePoolName)
```
Comment (thedadams) [inline pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go:307]: This message will be printed a bunch (not just on create). I think a comment here is sufficient. Maybe even combine this if-statement with the one above.

This behavior is something that should go in the docs and be release noted.

**Review by thedadams — CHANGES_REQUESTED** [review]

**Review by thedadams — APPROVED** [review]

File: pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go [hunk 2]
```diff
@@ -300,7 +300,8 @@ func machineDeployments(cluster *rancherv1.Cluster, capiCluster *capi.Cluster, d
 			machineDeploymentLabels[k] = v
 		}
 		machineSpecAnnotations := map[string]string{}
-		if !machinePool.DrainBeforeDelete {
+		// Ignore drain before delete etcd nodes
```
Comment (jiaqiluo) [inline pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go:303]: hmm, the comment is a bit confusing, could you rephrase it to be more clear? 
```suggestion
		// Ignore drain if DrainBeforeDelete is unset or the pool is for etcd nodes 
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Review by jiaqiluo — APPROVED** [review]

**Review by thedadams — APPROVED** [review]
